pkg/core/resource_graph.go

```diff
@@ -26,14 +26,19 @@ func (rg *ResourceGraph) AddResource(resource Resource) {
 	}
 }
 
-// AddDependency2 deliberately renamed from AddDependency in the short term to catch when dependencies were
-// being added in the inverse order. If either `source` or `dest` don't exist in the graph, they are added.
-func (rg *ResourceGraph) AddDependency2(source Resource, dest Resource) {
-	for _, res := range []Resource{source, dest} {
+// Adds a dependency such that `deployedFirst` has to be deployed before `deployedNext`. For example, if you have a
+// Lambda and its execution role, then:
+//
+//	╭───────────────╮   ╭────────────────╮
+//	│     first     ├──➤│      next      │
+//	│    IamRole    │   │ LambdaFunction │
+//	╰───────────────╯   ╰────────────────╯
+func (rg *ResourceGraph) AddDependency(deployedFirst Resource, deployedNext Resource) {
```

Comment:
This is flipping the order again.
`AddDependency(lambda, role)` is how it's used and results in the arrow `lambda -> role`

Reply:
Oh, I misread the `dag.AddDependency(role, resource)` in `execution_unit.go` (I had just opened it in the search preview, and didn't look to see what `resource` was — I assumed it was the lambda, but it's not.)

Reply:
The main thing I use for my mental model is the reflection-based one. All the edges are `dag.AddDependency(theStruct, theField)`

Reply:
I like that mental model. I just pushed a change to add it to the godocs (didn't use the reflection aspect, but the struct-and-field).